### ISSUE TITLE
About the SCANLoss - Reopening Issue # 24

### ISSUE THREAD

**Issue by rajat-tech-002** [wvangansbeke/Unsupervised-Classification#86]
@wvangansbeke 
Hi, 
![Screenshot from 2021-09-17 11-10-24](https://user-images.githubusercontent.com/46721220/133730425-f528ddd8-2cee-4905-94fd-f86b8a69fe1f.png)
`    # Total loss

       #  total_loss = consistency_loss - self.entropy_weight * entropy_loss`
The Entropy Weight is positive as mentioned in your paper. 
From the entropy() function, you are returning - (plogp) and then again you use -sign while calculation of total loss. 
That seems like here entropy weight = - 5.0. 
Can you please elaborate its significance? Thanks

![Screenshot from 2021-09-17 11-11-12](https://user-images.githubusercontent.com/46721220/133730523-e28ec1b7-bc54-447a-b49c-6500ad7dda39.png)

**Comment by wvangansbeke** [comment]
Hi @rajat-tech-002,

Entropy is `H(X) = -X*Log(X)`. This is also what the `entropy()` function calculates. We need to maximize the entropy, so we minimize `-H(X)` in the loss. The weight is always positive and simply balances both loss terms. This formulation was also used in the code, which makes it more clear what get's minimized/maximized imo.

**Comment by rajat-tech-002** [comment]
Thanks for the update.
